ipam/pool.go

```diff
@@ -515,7 +515,8 @@ func (ap *addressPool) requestAddress(address string, options map[string]string)
 		}
 
 		if ar == nil {
-			return "", errNoAvailableAddresses
+			err = errNoAvailableAddresses
+			return "", err
```

Comment:
What is the purpose of this change? style?

Reply:
Not a style thing. It's only so the deferd log statement above picks it up. It was confusing looking at the logs and seeing the function returned nil, nil when that wasn't true